torch/csrc/api/src/nn/modules/loss.cpp

```diff
@@ -32,5 +35,30 @@ Tensor HingeEmbeddingLossImpl::forward(
   return F::hinge_embedding_loss(input, target, options);
 }
 
+// ============================================================================
+
+TripletMarginLossImpl::TripletMarginLossImpl(
+    const TripletMarginLossOptions& options_)
+    : options(options_) {}
+
+void TripletMarginLossImpl::pretty_print(std::ostream& stream) const {
+  stream << "torch::nn::TripletMarginLoss(margin=" << options.margin() << 
+            ", p=" << options.p() <<
+            ", eps=" << options.eps() << std::boolalpha <<
+<<<<<<< HEAD
+            ", swap=" << options.swap() <<
+=======
+            ", swap=" <<options.swap() <<
+>>>>>>> Add C++ Frontend - triplet-margin-loss
+            ", reduction=" << options.reduction() << ")";
```

Comment:
It seems confusing that we have logic for printing `reduction` here, but the actual `pretty_print` string doesn't have `reduction`. I suggest removing the `reduction` printing logic here for now, until we have better support for printing named enums :)

Reply:
Hi @yf225 , I mistakenly pushed these changes. Sorry for wasting your time, I was still working on the PR.

Reply:
Here is my updated PR #27713 

Thank you